docs/data_classes.rst

```diff
@@ -0,0 +1,301 @@
+.. _data_classes:
+
+============
+Data Classes
+============
+ndcube provides its features via its data classes: `~ndcube.NDCube`, `~ndcube.NDCubeSequence` and `~ndcube.NDCollection`.  This section describes the purpose of each and how they are structured and instantiated.  To learn how to slicem visualize and perform coordinate transformations with these classes, see the :ref:`slicing`, :ref:`plotting` and :ref:`coordinates` sections.
+
+.. _ndcube:
+
+NDCube
+======
+ndcube's primary data class is `~ndcube.NDCube`.  It's designed for managing a single data array and set of WCS transformations.  `~ndcube.NDCube` provides unified slicing, visualization, coordinate conversion APIs as well as APIs for inspecting the data, coordinate transformations and metadata. `~ndcube.NDCube` does this in a way that is not specific to any number or physical type of axis.  It can therefore be used for any type of data (e.g. images, spectra, timeseries, etc.) so long as those data are represented by an array and a set of WCS transformations. Moreover, `~ndcube.NDCube` is agnostic to the fundamental array type in which the data is stored, as long as it behaves like a numpy array.
+Meanwhile, the WCS object can be any class, as long as it adhere's to the AstroPy `wcsapi (APE 14) <https://docs.astropy.org/en/stable/wcs/wcsapi.html>`_ specification. These features and flexibility make `~ndcube.NDCube` ideal for subclassing when creating tools for specific data types of data.  It enables developers and scientists to focus on developing the tools needed for their specific research while leveraging standarized APIs for non-data-type-specific functionalities (e.g. slicing).
+
+Initialize an NDCube
+--------------------
+To initialize the most basic `~ndcube.NDCube` object, we need is a `numpy.ndarray`-like array containing the data and an APE-14-compliant WCS object (e.g. `astropy.wcs.WCS`) describing the coordinate transformations to and from array-elements. Let's create a 3-D array of data with shape ``(3, 4, 5)`` where every value is 1.
+
+.. code-block:: python
+
+  >>> import numpy as np
+  >>> data = np.ones((3, 4, 5))
+
+Now let's create an `astropy.wcs.WCS` object.  Let the first world axis be wavelength, the second be helioprojective longitude, the third be helioprojective latitude. Remember that due to convention, the order WCS axes is reversed relative to the data array.  This means that the first two array axes will correspond to helioprojective latitude and longitude and the third to wavelength.
```

Comment:
I think it would be good to give a real world example of what this would be. For example, this could be a data cube which represents 5 images at 5 different wavelengths lalalala...